### PR TITLE
Add `nxos_vrf_global` resource module

### PR DESCRIPTION
**Pull request by roverflow**
##### SUMMARY
This PR adds the `nxos_vrf_global` module

##### ISSUE TYPE

- New Module Pull Request

### CODE REVIEW

File: plugins/module_utils/network/nxos/facts/vrf_global/vrf_global.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+from ansible_collections.cisco.nxos.plugins.module_utils.network.nxos.argspec.vrf_global.vrf_global import (
+    Vrf_globalArgs,
+)
+from ansible_collections.cisco.nxos.plugins.module_utils.network.nxos.rm_templates.vrf_global import (
+    Vrf_globalTemplate,
+)
+
+
+debugpy.listen(3000)
```
Comment (rohitthakur2590) [inline plugins/module_utils/network/nxos/facts/vrf_global/vrf_global.py:33]: This needs to be removed

File: plugins/modules/nxos_vrf_global.py [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+  returned: when I(state) is C(rendered)
+  type: list
+  sample:
+    - sample command 1
+    - sample command 2
+    - sample command 3
+gathered:
+  description: Facts about the network resource gathered from the remote device as structured data.
+  returned: when I(state) is C(gathered)
+  type: list
```
Comment (rohitthakur2590) [inline plugins/modules/nxos_vrf_global.py:372]: This needs to a type dict ; same for parsed operation.

File: plugins/modules/nxos_vrf_global.py [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+        the value is returned in the I(gathered) key within the result.
+      - The state I(parsed) reads the configuration from C(running_config) option and
+        transforms it into JSON format as per the resource module parameters and the
+        value is returned in the I(parsed) key within the result. The value of C(running_config)
+        option should be the same format as the output of command I(show running-config | section ^vrf).
+        connection to remote host is not required.
+    type: str
+"""
+
+EXAMPLES = """
```
Comment (rohitthakur2590) [inline plugins/modules/nxos_vrf_global.py:334]: Can we also populate the examples

File: plugins/modules/nxos_vrf_global.py [hunk 4]
```diff
@@ -0,0 +1,419 @@
...
+  sample:
+    - sample command 1
+    - sample command 2
+    - sample command 3
+rendered:
+  description: The provided configuration in the task rendered in device-native format (offline).
+  returned: when I(state) is C(rendered)
+  type: list
+  sample:
+    - sample command 1
```
Comment (rohitthakur2590) [inline plugins/modules/nxos_vrf_global.py:366]: please add relevant commands.

File: plugins/module_utils/network/nxos/config/vrf_global/vrf_global.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        """
+        wantd = {entry["name"]: entry for entry in self.want}
+        haved = {entry["name"]: entry for entry in self.have}
+
+        # if state is merged, merge want onto have and then compare
+        if self.state == "merged":
+            wantd = dict_merge(haved, wantd)
+
+        # if state is deleted, empty out wantd and set haved to wantd
+        if self.state == "deleted":
```
Comment (rohitthakur2590) [inline plugins/module_utils/network/nxos/config/vrf_global/vrf_global.py:75]: Where are we taking care of purged state operation ?

File: plugins/module_utils/network/nxos/config/vrf_global/vrf_global.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+        if self.state not in ["parsed", "gathered"]:
+            self.generate_commands()
+            self.run_commands()
+        return self.result
+
+    def generate_commands(self):
+        """Generate configuration commands to send based on
+        want, have and desired state.
+        """
+        wantd = {entry["name"]: entry for entry in self.want}
```
Comment (rohitthakur2590) [inline plugins/module_utils/network/nxos/config/vrf_global/vrf_global.py:67]: I don't think this would get us vrf list ?

**Review by rohitthakur2590 — CHANGES_REQUESTED** [review]

File: plugins/module_utils/network/nxos/facts/vrf_global/vrf_global.py (hunk 1 above)
Comment (roverflow) [reply plugins/module_utils/network/nxos/facts/vrf_global/vrf_global.py:33]: Will remove this

File: plugins/modules/nxos_vrf_global.py (hunk 3 above)
Comment (roverflow) [reply plugins/modules/nxos_vrf_global.py:334]: Sure, i have to start working on config side code this sprint, will update them

File: plugins/modules/nxos_vrf_global.py (hunk 4 above)
Comment (roverflow) [reply plugins/modules/nxos_vrf_global.py:366]: Will do!

File: plugins/module_utils/network/nxos/config/vrf_global/vrf_global.py (hunk 5 above)
Comment (roverflow) [reply plugins/module_utils/network/nxos/config/vrf_global/vrf_global.py:75]: Have not yet started with config side code

File: tests/integration/targets/nxos_vrf_global/vars/main.yaml [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+          vni_number: 5
+
+overridden:
+  commands: []
+  before:
+    - "{{ management[0] }}"
+    - description: testvrfhere
+      ip:
+        auto_discard: true
+        domain_name: redhat.com
```
Comment (NilashishC) [inline tests/integration/targets/nxos_vrf_global/vars/main.yaml:172]: please change to a RFC approved example domain name.

File: plugins/modules/nxos_vrf_global.py [hunk 8]
```diff
@@ -0,0 +1,1175 @@
...
+#   - ip route 192.0.0.0/24 192.0.2.22
+#   - ip route 192.0.0.0/24 192.0.2.22 vrf temp1
+#   - ip route 192.0.2.0/24 192.0.2.22 tag 2 4
+#   - multicast service-reflect interface Ethernet1/1 map interface Ethernet2/2
+#   - multicast service-reflect interface Ethernet2/1 map interface Ethernet4/2
+#   - ipv6 mld ssm-translate ff28::/16 2001:db8:0:abcd::2
+#   - ipv6 mld ssm-translate ff30::/16 2001:db8:0:abcd::5
+
+# after:
+# vrfs:
```
Comment (NilashishC) [inline plugins/modules/nxos_vrf_global.py:432]: indentation needs to be fixed

File: plugins/modules/nxos_vrf_global.py [hunk 9]
```diff
@@ -0,0 +1,1175 @@
...
+    config:
+      vrfs:
+        - name: test1
+    state: deleted
+
+# Task Output:
+# ------------
+#
+# before:
+# vrfs:
```
Comment (NilashishC) [inline plugins/modules/nxos_vrf_global.py:585]: indentation needs to be fixed

File: plugins/modules/nxos_vrf_global.py [hunk 10]
```diff
@@ -0,0 +1,1175 @@
...
+#   - no ip mroute 192.168.1.0/24 192.168.1.1
+#   - no ip mroute 192.168.1.0/24 192.168.1.2 2 vrf temp1
+#   - no ip multicast rpf select vrf temp1 group-list 238.1.0.0/24
+#   - no ip multicast rpf select vrf temp1 group-list 239.1.0.0/24
+#   - no ip route 192.0.0.0/24 192.0.2.22
+#   - no ip route 192.0.0.0/24 192.0.2.22 vrf temp1
+#   - no ip route 192.0.2.0/24 192.0.2.22 tag 2 4
+#
+# after:
+# vrfs:
```
Comment (NilashishC) [inline plugins/modules/nxos_vrf_global.py:667]: Are we actually rendering "vrfs" key under `after` and `before`?

**Review by NilashishC — CHANGES_REQUESTED** [review]
Please deprecate the nxos_vrf module and add a corresponding entry in changelog.

File: plugins/modules/nxos_vrf.py [hunk 11]
```diff
@@ -26,6 +26,10 @@
 extends_documentation_fragment:
 - cisco.nxos.nxos
 short_description: Manages global VRF configuration.
+deprecated:
+  alternative: nxos_vrf_global
+  why: Updated modules released with more functionality.
```
Comment (NilashishC) [inline plugins/modules/nxos_vrf.py:31]:
```suggestion
  why: Updated module released with more functionality.
```

Comment (roverflow) [reply plugins/modules/nxos_vrf.py]: had to remove this, this gives an error, just looked it up and seems like no collections have it, nxos has it but for very old modules only

**Review by rohitthakur2590 — APPROVED** [review]

**Review by NilashishC — APPROVED** [review]

**Pull request merged**